server/image_status.go

```diff
@@ -9,8 +9,20 @@ import (
 // ImageStatus returns the status of the image.
 func (s *Server) ImageStatus(ctx context.Context, req *pb.ImageStatusRequest) (*pb.ImageStatusResponse, error) {
 	logrus.Debugf("ImageStatus: %+v", req)
-	// TODO
-	// containers/storage will take care of this by looking inside /var/lib/ocid/images
-	// and getting the image status
-	return &pb.ImageStatusResponse{}, nil
+	image := ""
+	img := req.GetImage()
+	if img != nil {
+		image = img.GetImage()
+	}
+	status, err := s.images.ImageStatus(ctx, image)
```

Comment:
@nalind I believe k8s itself would have already pulled it. I'd say let's not worry about this issue _right now_ an defer it to when this PR is fully ready to be reviewed and integrated. As Mrunal pointed out there may be other issues, but would be simple to tackle each if we have this PR in a good shape to work on when merged.